enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,995 @@
+
+  // failureDomains holds the VSpherePlatformFailureDomainSpec which contains
+  // the definition of region, zone and the vCenter topology.
+  // If this is omitted failure domains (regions and zones) will not be used.
+  // +kubebuilder:validation:Optional
+  FailureDomains []VSpherePlatformFailureDomainSpec `json:"failureDomains,omitempty"`
+
+  // nodeNetworking defines the way that the Node IPs are selected
+  // from the addresses assigned to the Node in kube-api.
+  // If nodeNetwork is not provided, default selection
+  // will select the first address that is not a Localhost address.
+  // +kubebuilder:validation:Optional
+  NodeNetworking VSpherePlatformNodeNetworking `json:"nodeNetworking,omitempty"`
+}
+```
+
+### vSphere Machine - VSphereMachineProviderSpec change
+
+Including the change to the vSphere Infrastructure and install platform
+spec we need to include a new parameter to the 
+`VSphereMachineProviderSpec` - `FailureDomain`. This
+string will associate the Machine to the FailureDomain that is defined
+via the install-config platform spec.
+
+```golang
+type VSphereMachineProviderSpec struct {
+
+  // ...
+
+	// CloneMode specifies the type of clone operation.
+	// The LinkedClone mode is only support for templates that have at least
+	// one snapshot. If the template has no snapshots, then CloneMode defaults
+	// to FullClone.
+	// When LinkedClone mode is enabled the DiskGiB field is ignored as it is
+	// not possible to expand disks of linked clones.
+	// Defaults to LinkedClone, but fails gracefully to FullClone if the source
+	// of the clone operation has no snapshots.
+	// +optional
+	CloneMode CloneMode `json:"cloneMode,omitempty"`
+
+
+	// FailureDomain is the failure domain unique identifier this Machine should be attached to, as defined in Infrastructure spec.
+	// For this infrastructure provider, the name is equivalent to the name of the VSphereDeploymentZone.
+	FailureDomain *string `json:"failureDomain,omitempty"`
+}
+```
+
+### CCCMO
+
+The CCCMO will need to be modified to support the VSpherePlatformSpec.
+It will need to be able to use the existing INI-based cloud-config spec
+and also translate that configuration to YAML-based cloud-config.
+
+### installer
+
+#### Platform Spec
+
+The platform spec needs to be modified to support our initial goals of multiple
+datacenters (regions) and clusters (zones) and vcenters. This platform spec
+design is based on the changes suggested in openshift/api, Cluster API for vSphere
+and out-of-tree CCM.
+
+We are adding multiple additional parameters to the Platform struct:
+
+- `VCenters`
+- `DeploymentZones`
+- `FailureDomains`
+
+`VCenters` will contain the connections and configuration for each vCenter
+that is required for the out-of-tree CCM. Note: Only a single vCenter will
+be supported by this effort. While the out-of-tree CCM supports multiple vCenters
+the out-of-tree CSI does not.
```

Comment:
Related to [my other comment ](https://github.com/openshift/enhancements/pull/918/files#r942841375), if only one `vCenter` is supported, why do we need the new `vCenters`? I'm not opposed to the new `vCenter`, but I do think if we are going to introduce `vCenters` we need to have a solid deprecation plan for `vCenter`. If we're not ready for that, perhaps we should consider whether we can stick with `vCenter`. We have deprecated a few fields recently... but not the most important field for our most popular platform...